semgrep/semgrep/core_runner.py

```diff
@@ -253,49 +256,98 @@ def _run_core_command(
                 self._write_equivalences_file(equiv_file, equivalences)
                 cmd += ["-equivalences", equiv_file.name]
 
-            if report_time:
+            if self._report_time:
                 cmd += ["-json_time"]
 
+            if self._output_settings.debug or self._output_settings.verbose_errors:
+                cmd += ["-debug"]
+
             core_run = sub_run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            logger.debug(core_run.stderr.decode("utf-8", errors="replace"))
-
-            out_bytes = core_run.stdout
-            err_bytes = core_run.stderr
-            returncode = core_run.returncode
-            if returncode != 0:
-                output_json = self._parse_core_output(out_bytes, err_bytes, returncode)
-
-                if "error" in output_json:
-                    self._raise_semgrep_error_from_json(output_json, patterns, rule)
-                else:
-                    raise SemgrepError(
-                        f"unexpected json output while invoking semgrep-core with rule '{rule.id}':\n{PLEASE_FILE_ISSUE_TEXT}"
-                    )
+            output_json = self._extract_core_output(rule, patterns, core_run)
+            return output_json
 
-            output_json = self._parse_core_output(out_bytes, err_bytes, returncode)
+    def _extract_core_output(
+        self, rule: Rule, patterns: List[Pattern], core_run: subprocess.CompletedProcess
+    ) -> Dict[str, Any]:
+        logger.debug(core_run.stderr.decode("utf-8", errors="replace"))
+
+        semgrep_output = core_run.stdout.decode("utf-8", errors="replace")
+        semgrep_error_output = core_run.stderr.decode("utf-8", errors="replace")
+
+        # Always print semgrep-core's error output, which includes
+        # semgrep-core's logging if it was requested.
+        if semgrep_error_output != "":
+            name = f"[process {os.getpid()}, rule '{rule.id}']"
+            print(
```

Comment:
let's. use logger class so this goes to stderr and respects verbosity
```suggestion
            logger.debug(
```